ark/utils/deepcell_service_utils.py

```diff
@@ -175,11 +184,50 @@ def run_deepcell_direct(input_dir, output_dir, host='https://deepcell.org',
         }
         f.seek(0)
 
-    upload_response = requests.post(
-        upload_url,
-        timeout=timeout,
-        files=upload_fields
-    ).json()
+    # define and mount a retry instance to call the Deepcell API again if needed
+    retry_strategy = Retry(
+        total=num_retries,
+        status_forcelist=[404, 500, 502, 503, 504],
+        method_whitelist=['HEAD', 'GET', 'POST', 'PUT', 'DELETE', 'OPTIONS', 'TRACE']
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+
+    http = requests.Session()
+    http.mount('https://', adapter)
+    http.mount('http://', adapter)
+
+    total_retries = 0
+    while total_retries < num_retries:
+        # handles the case if the main endpoint can't be reached
+        print("total number of retries is {}".format(total_retries))
+        try:
+            upload_response = http.post(
+                upload_url,
+                timeout=timeout,
+                files=upload_fields
+            )
+        except RetryError as re:
+            print("Failed to reached Deepcell after %d attempts: "
+                  "the server is likely down" % num_retries)
+            return 1
+
+        # handles the case if the endpoint returns an invalid JSON
+        # indicating an internal API error
+        try:
+            upload_response = upload_response.json()
+        except JSONDecodeError as jde:
+            total_retries += 1
+            print("retrying, total is {}".format(total_retries))
+            continue
+
+        # if we reach the end no errors were encountered on this attempt
+        break
+
+    # if the JSON could not be decoded num_retries number of times
+    if total_retries == num_retries:
+        print("Unable to process Deepcell response after %d attempts: "
```

Comment:
The only way this line of code gets reached if it's a JSON decode error. Saying this is a server error is just confusing

Reply:
The JSON decode error happens because of a 502 error (or similar) that happens internally and gets returned in HTML format in the response. We could instead inform the user that there was a problem decoding the response from Deepcell and to check if the website is down. Let me know what you think.

Reply:
Yeah, saying there was an error decoding the json sounds good.